src/main/java/rcms/utilities/daqexpert/persistence/LogicModuleRegistry.java

```diff
@@ -60,6 +61,12 @@
 	FlowchartCase4(new FlowchartCase4(), ConditionGroup.FLOWCHART, "", 10007),
 	FlowchartCase5(new FlowchartCase5(), ConditionGroup.FLOWCHART, "", 10008),
 	FlowchartCase6(new FlowchartCase6(), ConditionGroup.FLOWCHART, "", 10009),
+	
+	// note that this must be declared after all other logic modules
+	// identifying an error condition in order to have UnidentifiedFailure
+	// run after the others
+	UnidentifiedFailure(new UnidentifiedFailure(), ConditionGroup.OTHER, "", 19999),
```

Comment:
Actually you have to add it after all modules (including comparators). The problem arises when we will add more LM, which we would have to add after UnidentifiedFailure due to db id based on oridinal no of enum. This will result in running UnidentifiedFailure run before new LM. I will think of solution next week. Have a great weekend!